app/assets/javascripts/utils/articles_or_items_utils.js

```diff
@@ -0,0 +1,14 @@
+
+export default class ArticlesOrItemsUtils {
+// Returns article or item, based on home Wiki.
+  static articlesOrItems(project) {
+    return project === 'wikidata' ? 'items' : 'articles';
+  }
+
+// Returns article or item message.
+  static articlesOrItemsI18n(messageKey, articlesOrItems, defaultarticlesOrItems = 'articles') {
```

Comment:
I suggest having this take the `project` as a parameter instead of `articlesOrItems`, and use the articlesOrItems function internally, so that the components that use it only need one function rather than both.

I suggest calling the class `ArticleUtils` and the function just `i18n`, to make it a little more concise in all the places that will use it. I think `ArticleUtils.i18n` communicates well enough the purpose of providing appropriate terminology for pages of different types, in parallel to `CourseUtils.i18n`.